### PR TITLE
#1873: Page reloads with login opened

### PR DESCRIPTION
**Pull request by martinb14**
http://platform.dandigbib.org/issues/1873

### CODE REVIEW

**Comment by cableman** [comment]
Please see the issue for more information.

File: modules/p2/ding_entity_rating/ding_entity_rating.module [hunk 1]
```diff
@@ -76,28 +76,31 @@ function ding_entity_rating_access($account) {
...
+  $commands = array();
+  
+  // Grabbed from ding_reservation_reserve_ajax()
+  if (!user_is_logged_in()) {
+    // Trigger log-in (the reservation link will be triggered on success).
+    $commands[] = ajax_command_ding_user_authenticate('');
+  }
+  elseif (!ding_user_is_provider_user($user)) {
+    // Error not library user.
+    $commands[] = ajax_command_ding_popup('ding_reservation', t('Error'), '<p>' . t('Only library user can make reservations.') . '</p>');
```
Comment (JrgenGNielsen) [inline modules/p2/ding_entity_rating/ding_entity_rating.module]: Shouldn't this be something like 'Only library users can make ratings'?

File: modules/p2/ding_entity_rating/ding_entity_rating.module [hunk 2]
```diff
@@ -76,28 +76,31 @@ function ding_entity_rating_access($account) {
...
   }
   else {
-    // Only ajax supported.
-    return '';
+    ding_entity_rating_add_to_list($entity, $user, $rating);
+    
+    $commands[] = ajax_command_ding_popup(
+      'ding_reservation',
+      t('Rated'), 
+      '<p>' . theme('ding_entity_rating_thank_you') . '</p>'
```
Comment (JrgenGNielsen) [inline modules/p2/ding_entity_rating/ding_entity_rating.module]: calling the theme() function directly is strongly discouraged - see: https://api.drupal.org/api/drupal/includes!theme.inc/function/theme/7.x

File: modules/p2/ding_entity_rating/js/ding_entity_rating.js [hunk 3]
```diff
@@ -65,31 +65,35 @@
...
-      $.get('/ding_entity_rating/ajax/' + _path + '/' + _index, function(responseText) {
-        $this.parent().append(responseText);
-        $('.ding-entity-rating-respons', $this.parent()).delay(5000).fadeOut();
-      });
+      
+      var $this = $(this),
+        _path = $this.parent().attr('data-ding-entity-rating-path'),
+        _index = $this.index() + 1,
+        
+        url = '/ding_entity_rating/' + _path + '/' + _index,
```
Comment (JrgenGNielsen) [inline modules/p2/ding_entity_rating/js/ding_entity_rating.js]: url should start with settings.basePath + '/ding_entity_rating/' + [...]

File: modules/p2/ding_entity_rating/ding_entity_rating.module (hunk 1 above)
Comment (martinb14) [inline modules/p2/ding_entity_rating/ding_entity_rating.module]: thx, fixed.

File: modules/p2/ding_entity_rating/ding_entity_rating.module (hunk 2 above)
Comment (martinb14) [inline modules/p2/ding_entity_rating/ding_entity_rating.module]: thx, fixed.

**Comment by JrgenGNielsen** [comment]
reviewed and accepted

**Pull request merged**